dali/imgcodec/decoders/nvjpeg_lossless/nvjpeg_lossless_test.cc

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/imgcodec/decoders/nvjpeg_lossless/nvjpeg_lossless.h"
+#include <gtest/gtest.h>
+#include <limits>
+#include <string>
+#include "dali/imgcodec/decoders/decoder_test_helper.h"
+#include "dali/imgcodec/parsers/jpeg.h"
+#include "dali/test/dali_test.h"
+#include "dali/test/dali_test_config.h"
+
+namespace dali {
+namespace imgcodec {
+namespace test {
+
+namespace {
+
+std::string data_path(const std::string &relative_path) {
+  return make_string_delim('/', testing::dali_extra_path(), "db/single/jpeg_lossless",
+                           relative_path);
+}
+std::string reference_path(const std::string &relative_path) {
+  return make_string_delim('/', testing::dali_extra_path(), "db/single/reference/jpeg_lossless",
+                           relative_path);
+}
+
+
+struct ImageBuffer {
+  std::vector<uint8_t> buffer;
+  ImageSource src;
+
+  explicit ImageBuffer(const std::string &path) {
+    std::ifstream stream(path, std::ios::binary);
+    buffer = {std::istreambuf_iterator<char>(stream), std::istreambuf_iterator<char>()};
+    src = ImageSource::FromHostMem(buffer.data(), buffer.size());
+  }
+};
```

Comment:
It's the 5th copy of this class and about a time we moved it to some kind of imgcodec_test_utils.h or similar (not in `dali/imgcodec/`, not `include/dali/imgcodec`).